src/main/java/org/symphonyoss/integration/webhook/jira/parser/v2/CommentMetadataParser.java

```diff
@@ -0,0 +1,199 @@
+   * comment action (add, edit, delete)
+   * @param input The root json node
+   */
+  private void processCommentAction(JsonNode input) {
+    String webHookEvent = input.path(ISSUE_EVENT_TYPE_NAME).asText();
+    ObjectNode commentNode = (ObjectNode) input.with(COMMENT_PATH);
+    if (commentNode != null) {
+      commentNode.put(ACTION_ENTITY_FIELD, actions.get(webHookEvent));
+    }
+  }
+
+  /**
+   * This method changes the self link sent by Jira into a common comment Jira url
+   * @param input The root json node
+   */
+  private void processCommentLink(JsonNode input) {
+    ObjectNode commentNode = getCommentNode(input);
+
+    if (commentNode != null) {
+      JsonNode issueNode = input.path(ISSUE_PATH);
+      String linkedIssueField = getLinkedIssueField(issueNode);
+      String linkedCommentLink = getLinkedCommentLink(linkedIssueField, issueNode);
+      commentNode.put(LINK_ENTITY_FIELD, linkedCommentLink);
+    }
+  }
+
+  /**
+   * This method builds the comment permalink according to the issue link and the comment node,
+   * i.e., the given issue link must be built based on the issue related to the given comment node
+   * @param issueLink
+   * @param commentNode
+   * @return
+   */
+  private String getLinkedCommentLink(String issueLink, JsonNode commentNode) {
+    String commentId = commentNode.path(ID_PATH).asText();
+    StringBuilder commentLink = new StringBuilder(issueLink);
+
+    if (!StringUtils.isEmpty(commentId)) {
+      commentLink.append("?");
+      commentLink.append(String.format(COMMENT_LINK_SUFFIX, commentId, commentId));
+    }
+
+    return commentLink.toString();
+  }
+
+  /**
+   * This searches through the comment body and replaces
+   * @param input
+   */
+  private void processCommentMentions(JsonNode input) {
+    ObjectNode commentNode = getCommentNode(input);
+
+    if (commentNode != null) {
+      String comment = commentNode.path(BODY_PATH).asText();
+
+      Map<String, User> userMentions = determineUserMentions(comment);
+      if (userMentions != null && !userMentions.isEmpty()) {
+        for (Map.Entry<String, User> userEntry : userMentions.entrySet()) {
+          User user = userEntry.getValue();
+
+          //FIXME Agent v3 doen't support tags inside metadata
+//          String userMention =
+//              ParserUtils.presentationFormat(MESSAGEML_MENTION_EMAIL_FORMAT, user
+// .getEmailAddress())
+//                  .toString();
+          String userMention = user.getDisplayName();
+          comment = comment.replaceAll("\\[~" + userEntry.getKey() + "]", userMention);
+        }
+      }
+
+      comment = JiraParserUtils.stripJiraFormatting(comment);
+      commentNode.put(BODY_PATH, comment);
+    }
+  }
+
+  private ObjectNode getCommentNode(JsonNode input) {
+    return input.hasNonNull(COMMENT_PATH) ? (ObjectNode) input.path(COMMENT_PATH) : null;
+  }
+
+  private Map<String, User> determineUserMentions(String comment) {
```

Comment:
This method has a userService call, which needs the integration user information, which is filled by the factory. Create a helper with this information as a parameter seems strange, thus I haven't found a common place to put this code.
Since the v1 support will be discontinued soon, I think we can leave this as is for now